metagraph/src/graph/representation/canonical_dbg.hpp

```diff
@@ -22,43 +22,29 @@ class CanonicalDBG : public DeBruijnGraph {
     /**
      * Constructs a CanonicalDBG
      * @param graph a graph
-     * @param primary indicates whether the underlying graph is primary or not
-     * (i.e., only one of a k-mer and its reverse complement are in the graph)
      * @param cache_size the number of graph traversal call results to be cached
      */
-    CanonicalDBG(const DeBruijnGraph &graph,
-                 bool primary = false,
-                 size_t cache_size = 100'000);
+    CanonicalDBG(const DeBruijnGraph &graph, size_t cache_size = 100'000);
 
     /**
      * Constructs a CanonicalDBG
      * @param graph a graph
-     * @param primary indicates whether the underlying graph is primary or not
-     * (i.e., only one of a k-mer and its reverse complement are in the graph)
      * @param cache_size the number of graph traversal call results to be cached
      */
-    CanonicalDBG(DeBruijnGraph &graph, bool primary = false, size_t cache_size = 100'000);
+    CanonicalDBG(DeBruijnGraph &graph, size_t cache_size = 100'000);
 
     /**
      * Constructs a CanonicalDBG
      * @param graph a pointer to the graph
-     * @param primary indicates whether the underlying graph is primary or not
-     * (i.e., only one of a k-mer and its reverse complement are in the graph)
      * @param cache_size the number of graph traversal call results to be cached
      */
-    CanonicalDBG(std::shared_ptr<const DeBruijnGraph> graph,
-                 bool primary = false,
-                 size_t cache_size = 100'000);
+    CanonicalDBG(std::shared_ptr<const DeBruijnGraph> graph, size_t cache_size = 100'000);
     /**
      * Constructs a CanonicalDBG
      * @param graph a pointer to the graph
-     * @param primary indicates whether the underlying graph is primary or not
-     * (i.e., only one of a k-mer and its reverse complement are in the graph)
      * @param cache_size the number of graph traversal call results to be cached
      */
-    CanonicalDBG(std::shared_ptr<DeBruijnGraph> graph,
-                 bool primary = false,
-                 size_t cache_size = 100'000);
+    CanonicalDBG(std::shared_ptr<DeBruijnGraph> graph, size_t cache_size = 100'000);
```

Comment:
it would be good to benchmarks the speed for different cache size at some point.

Maybe it never makes much sense to make the cache size larger than 100k, for example, and then we can simply set that as a global constant.

Reply:
Instead of keeping this fixed, I set it up such that `--batch-size` also sets the size of the cache. I still have to test this out on larger graphs, but it seems to work well for the microbe 750 subset.

Reply:
Maybe having the cache size of 1 or 10 MB is enough and increasing it never makes things faster.

In this case, we could hard code it to that value.

Reply:
I'm playing around with different values now, so far on a 5MB input, it seems like 100KB and 1MB are optimal. I'll see if that's still the case with a larger file.

Reply:
There are diminishing returns past 500KB, but that could just be because of my particular input

Reply:
I created a batching scheme where if the given batch size is too big (the entire query fits in the batch), then it splits the batch into `get_num_threads()` mini batches